csrc/scheduler/utils.cpp

```diff
@@ -473,12 +473,38 @@ class PersistentBufferResolution : public IterVisitor {
 
 } // namespace
 
-// Returns all broadcast tvs from the given vector of Vals.
-std::vector<TensorView*> getBroadcastTvs(const std::vector<Val*>& dep_vals) {
+// Returns all broadcast tvs whose produers are reduction tvs and each broadcast
+// dim is mapped to a reduction dim.
+std::vector<TensorView*> getBufferProjectableBroadcastsTvs(
+    const std::vector<Val*>& dep_vals,
+    const TensorView* persistent_buffer) {
   std::vector<TensorView*> broadcast_tvs;
+  broadcast_tvs.reserve(dep_vals.size());
   for (auto val : dep_vals) {
     if (auto tv = dynamic_cast<TensorView*>(val)) {
-      if (tv->hasBroadcast()) {
+      if (!tv->definition()->isA<BroadcastOp>() || tv == persistent_buffer) {
+        continue;
+      }
+      const auto& producers = ir_utils::producerTvsOf(tv);
+      NVF_ERROR(
+          producers.size() == 1,
+          "Tv defined by BroadcastOp must have one produer! Got: ",
+          producers.size());
+      auto producer = producers.at(0);
+      if (!producer->definition()->isA<ReductionOp>()) {
```

Comment:
The scheduler already knows which tensors are the reduction outputs of the normalization pattern. Shouldn't we check this producer is one of them?

Reply:
Changed the algorithm a little bit. For each buffer, will get all the reduction tvs it depends on, then for each reduction tv, get the broadcast tv which broadcasts the reduction dims. So, if the buffer depends on `N reduction tvs`, there must be `N broadcast tvs`, otherwise, not allowed to project.